src/engine/pieces/piece.ts

```diff
@@ -13,8 +17,35 @@ export default class Piece {
         throw new Error('This method must be implemented, and return a list of available moves');
     }
 
+    public addAvailableMoveAndContinue(availableMoves: Array<Square>, newRow : number, newCol : number, currentPosition : Square, board : Board){
+        let newPos = new Square(newRow, newCol);
+        let newPosInBound = newPos.inBoundsCheck();
+        if(!currentPosition.equals(newPos) && newPosInBound){
+            let pieceAtNewPos = board.getPiece(newPos);
+            if(pieceAtNewPos == undefined || (pieceAtNewPos.player != this.player && !pieceAtNewPos.isKing)){
```

Comment:
I'd maybe split this 
`pieceAtNewPos == undefined || (pieceAtNewPos.player != this.player && !pieceAtNewPos.isKing`
into a separate nicely named function to make it clearer what is happening

Reply:
Fixed